hazelcast/src/main/java/com/hazelcast/util/JVMUtil.java

```diff
@@ -36,12 +36,19 @@
      * Defines the costs for a reference in Bytes.
      */
     @SuppressWarnings("checkstyle:magicnumber")
-    public static final int REFERENCE_COST_IN_BYTES = isCompressedOops() ? 4 : 8;
+    public static final int REFERENCE_COST_IN_BYTES = is32bitJVM() || isCompressedOopsOrNull() ? 4 : 8;
```

Comment:
should this be `isCompressedOops` instead of `isCompressedOopsOrNull `? Maybe we can think a better name for `isCompressedOopsOrNull` to prevent confusion.